docs/Developers/Debug_Bears.rst

```diff
@@ -0,0 +1,78 @@
+Debug Bears
+===========
+
+This document provide a brief overview of coala's Debugging Interface.
+The Debugging Interface will help users to debug Bear code and step through
+it using the pdb interface.
+
+Details
+-------
+
+After invoking coala's debugger it will step into the ``run()`` method of
+bear regardless of using ``yield`` or not and will step out as soon as bear
+finish analysis.
+
+.. note::
+
+    Pdb's quit-command (``q``) has been remapped so coala continues its
+    normal execution without aborting. So, ``quit`` or `q` will first clear
+    all the breakspoints then continue ( ``c`` or ``continue`` ) execution.
+
+Usage of coala's Debugger
+=========================
+
+Command Line Interface
+----------------------
+
+Users can specify the bear they want to debug using ``--debug-bears``, i.e.
+
+.. code:: shell
+
+    $coala -b PEP8Bear,HelloWorldBear -f <filename> --debug-bears HelloWorldBear
+
+Below is the code given for a simple ``HelloWorldBear`` that prints a debug
+message for each file:
+
+.. code:: python
+
+    import logging
+
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class HelloWorldBear(LocalBear):
+        def run(self, filename, file):
+            logging.debug('Hello World! Checking file {}.'.format(filename))
```

Comment:
yield or return something. I believe I already mentioned that I consider bears returning `None` to be invalid / buggy bears :) The nextgen core will enforce that bugginess-attitude for bears that don't return iterables :)

Reply:
And maybe move it to the paragraph above, and maybe "simulate" a debug session, by explaining where the bear is exactly entered, and how it steps through code. This example one is not immediately related to the usage via CLI :+1:

Reply:
I think moving it above will become a mess as I am telling how to enable debugger on `HelloWorldBear` through CLI i.e., 
`$coala -b PEP8Bear,HelloWorldBear -f <filename> --debug-bears HelloWorldBear` but If I moved the code of `HelloWorldBear` to above para then I first have to tell how to enable debugger and then tell how it will work or move through it.

I will add more detail below :+1: